### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Jose76Tron**
-Compiling /lib/tools with OpenJDK 8 without JAVA_HOME
-Teleporting others permission, fixes /tp (player) (player)
-Fixes using empty item frames as shop and selling/buying Air.
-Added "fe.signs.usesigncommands" permission
-Fixed [command] signs not working

### CODE REVIEW

File: src/main/java/com/forgeessentials/signtools/SignToolsModule.java [hunk 1]
```diff
@@ -120,35 +117,45 @@ public void onPlayerInteract(PlayerInteractEvent event)
 
             }
 
-            String[] signText = getFormatted(sign.signText);
+            String[] signText = getUnFormatted(sign.signText);
```
Comment (spacebuilder2020) [inline src/main/java/com/forgeessentials/signtools/SignToolsModule.java:120]: What is the reason for this change?

File: src/main/java/com/forgeessentials/signtools/SignToolsModule.java [hunk 2]
```diff
@@ -120,35 +117,45 @@ public void onPlayerInteract(PlayerInteractEvent event)
...
-                        FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager().executeCommand(event.getEntityPlayer(), send);
-                        event.setCanceled(true);
+                        String send = signText[1] + " " + signText[2] + " " + signText[3];
+                        if (send != null && FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager() != null)
+                        {
+                            FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager().executeCommand(event.getEntityPlayer(), send);
+                            event.setCanceled(true);
+                        }
                     }
+                    else
```
Comment (spacebuilder2020) [inline src/main/java/com/forgeessentials/signtools/SignToolsModule.java:145]: Code Style rules for Forge Essentials requires brackets for every if / else even if it is a one liner.

File: src/main/java/com/forgeessentials/signtools/SignToolsModule.java [hunk 3]
```diff
@@ -120,35 +117,45 @@ public void onPlayerInteract(PlayerInteractEvent event)
...
 
     }
 
-    private String[] getFormatted(ITextComponent[] text)
+    private String[] getUnFormatted(ITextComponent[] text)
     {
         String[] out = new String[text.length];
         for (int i = 0; i < text.length; i++) {
-            out[i] = text[i].getFormattedText().replace(ChatOutputHandler.COLOR_FORMAT_CHARACTER, '&');
+            out[i] = text[i].getUnformattedText().replace(ChatOutputHandler.COLOR_FORMAT_CHARACTER, '&');
```
Comment (spacebuilder2020) [inline src/main/java/com/forgeessentials/signtools/SignToolsModule.java:160]: Again, why is this changing to unformatted text.  What do you wish to solve by doing this?

File: build.gradle [hunk 4]
```diff
@@ -201,6 +201,9 @@ dependencies {
 	// TOOLS.jar from jdk for javadoc support
     // compile files("$jdkHome/lib/tools.jar")
     compile files(System.getenv("JAVA_HOME") + "/lib/tools.jar")
+	dependencies {
```
Comment (spacebuilder2020) [inline build.gradle]: This doesn't look right.  There shouldn't be a nested dependency block...

Also, what is the reason for changing the path?

**Review by spacebuilder2020 — CHANGES_REQUESTED** [review]
Several things must be addressed before this can be considered

File: build.gradle (hunk 4 above)
Comment (Jose76Tron) [reply build.gradle]: On linux default JAVA_HOME env path is not set when you install openjdk 8, sorry for the nested dependency, fixing it

File: src/main/java/com/forgeessentials/signtools/SignToolsModule.java (hunk 2 above)
Comment (Jose76Tron) [reply src/main/java/com/forgeessentials/signtools/SignToolsModule.java:145]: Didn't know, sorry

**Comment by spacebuilder2020** [comment]
@Jose76Tron Can you please be more clear on why you changed the signText to use formatted text instead of unformatted text?

**Comment by spacebuilder2020** [comment]
@Jose76Tron Can you please explain the reasoning for changing from FormattedText to Unformatted text.  The unformatted text is needed so that color information is passed to the sign click event.  I understand it has something to do with [command] but what exactly is broken?
